pkg/platform/runtime/setup/setup.go

```diff
@@ -163,7 +170,15 @@ func (s *Setup) update() error {
 	if err != nil {
 		return locale.WrapError(err, "err_stored_artifacts", "Could not unmarshal stored artifacts, your install may be corrupted.")
 	}
-	s.deleteOutdatedArtifacts(changedArtifacts, storedArtifacts)
+	if buildResult.BuildEngine == model.Camel {
+		// for camel builds we have to wipe previous installations
+		os.RemoveAll(s.store.InstallPath())
+	} else {
+		err := s.deleteOutdatedArtifacts(changedArtifacts, storedArtifacts)
+		if err != nil {
+			return locale.WrapError(err, "err_delete_outdated", "Could not remove outdated artifacts files in {{.V0}}.  You may try to remove the entire directory manually.", s.store.InstallPath())
+		}
+	}
```

Comment:
This should be happening in the camel package. Add an interface method if you need to, I don't want to start sprinkling this into the non-builder specific code.

The ideal of keeping that interface as light as possible does not mean we should cheat our way around it.